### ISSUE TITLE
regression in 2.7.0: false positive consider-using-generator

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#4132]
### Steps to reproduce

```python
def hashable(x):
    "Turn the object into something that can be inserted into a set/dict."
    # we could add sorted() before tuple, but that would break heterogeneous
    # collections, e.g., ["a",1] and {"a":1,2:"b"}
    if isinstance(x, dict):
        return tuple([(hashable(k),hashable(v)) for k,v in x.items()])
    if isinstance(x, (list, tuple)):
        return tuple(map(hashable, x))
    if isinstance(x, set):
        return frozenset(x)
    return {x}.pop()            # fails if x is not hashable
```

### Current behavior
Diagnostics:
```
[R1728(consider-using-generator), hashable] Consider using a generator instead 'tuple((hashable(k), hashable(v)) for (k, v) in x.items())'
```

### Expected behavior

I don't see how to use a generator here, so no diagnostics is expected (and none was issued in 2.6.0)

### pylint --version output
```
pylint 2.7.0
astroid 2.5
Python 3.9.2 (default, Feb 19 2021, 17:09:53) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by sam-s** [comment]
cf. https://github.com/PyCQA/pylint/issues/4133

**Comment by ikraduya** [comment]
Please refer to this #3309
Especially at [this](https://github.com/PyCQA/pylint/pull/3309#discussion_r577160339) discussion 

TL;DR: removing "[ ]" inside `tuple` function should be considered since list comprehension will make unnecessary data copies. It will be very useful if you are working with really long lists or sets, if not you can ignore the message.
